packages/@react-aria/landmark/test/useLandmark.test.tsx

```diff
@@ -606,7 +610,88 @@ describe('LandmarkManager', function () {
       </div>
     );
 
-    let spyWarn = jest.spyOn(console, 'warn').mockImplementation(() => {});
     expect(spyWarn).toHaveBeenCalledWith('Page contains more than one landmark with the \'navigation\' role and \'First nav\' label. If two or more landmarks on a page share the same role, they must have unique labels.');
   });
+
+  it('Should not navigate to a landmark that has been removed from the DOM', function () {
+    let tree = render(
+      <div>
+        <Navigation>
+          <ul>
+            <li><a href="/home">Home</a></li>
+            <li><a href="/about">About</a></li>
+            <li><a href="/contact">Contact</a></li>
+          </ul>
+        </Navigation>
+        <Region>
+          <TextField label="First Name" />
+        </Region>
+        <Main>
+          <TextField label="Last Name" />
+        </Main>
+      </div>
+    );
+
+    fireEvent.keyDown(document.activeElement, {key: 'F6'});
+    fireEvent.keyUp(document.activeElement, {key: 'F6'});
+    expect(document.activeElement).toBe(tree.getAllByRole('link')[0]);
+
+    fireEvent.keyDown(document.activeElement, {key: 'F6'});
+    fireEvent.keyUp(document.activeElement, {key: 'F6'});
+    expect(document.activeElement).toBe(tree.getAllByRole('textbox')[0]);
+
+    fireEvent.keyDown(document.activeElement, {key: 'F6'});
+    fireEvent.keyUp(document.activeElement, {key: 'F6'});
+    expect(document.activeElement).toBe(tree.getAllByRole('textbox')[1]);
+
+
+    tree.getByRole('region').remove();
```

Comment:
this is unlikely happen in a real react app, instead, let's call rerender in the opposite manner to how you do in the next test to add a landmark

```suggestion
    rerender(
      <div>
        <Navigation>
          <ul>
            <li><a href="/home">Home</a></li>
            <li><a href="/about">About</a></li>
            <li><a href="/contact">Contact</a></li>
          </ul>
        </Navigation>
        <Main>
          <TextField label="Last Name" />
        </Main>
      </div>
    );
```

Reply:
Thanks. Couldn't find any examples where we did something similar. That makes sense.